File: evidence/generated-systems/golang/fintech-on-golang/fintech-on-golang/internal/model/FinancialInstitution.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// FinancialInstitution Declaration
//==============================================================
type FinancialInstitution struct {
    gorm.Model
     Name                                    string
    LegalName                                    string
    CountryOfIncorporation                                    string
    Bic                                                            string
    Website                                    string
     Branches           []Branch `gorm:"foreignKey:BranchesFromFinancialInstitutionId"`
     Customers           []Customer `gorm:"foreignKey:CustomersFromFinancialInstitutionId"`
     ProductOfferings           []ProductOffering `gorm:"foreignKey:ProductOfferingsFromFinancialInstitutionId"`
     PaymentProcessors           []PaymentProcessor `gorm:"foreignKey:PaymentProcessorsFromFinancialInstitutionId"`
     CompliancePolicies           []CompliancePolicy `gorm:"foreignKey:CompliancePoliciesFromFinancialInstitutionId"`

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/fintech-on-rails/fintechonrails/app/models/financialinstitution.rb
class FinancialInstitution < ApplicationRecord


  composed_of :bIC,
    class_name: "BIC",
    mapping: [
      %w[bIC_value value]
    ]

  has_many :Branches, class_name: 'Branch'
  has_many :Customers, class_name: 'Customer'
  has_many :ProductOfferings, class_name: 'ProductOffering'
  has_many :PaymentProcessors, class_name: 'PaymentProcessor'
  has_many :CompliancePolicies, class_name: 'CompliancePolicy'

end
